backend/src/auth/services/auth_service.py

```diff
@@ -1,82 +1,225 @@
+    InvalidPassword,
+    InvalidUserData,
+    RegisterFailed,
+    TokenDoesNotExist,
+    UserDoesNotExist,
+    UsernameTaken,
+)
+from src.users.models.user_model import User
+from src.users.repositories import UserRepository
+from src.users.schemas.profile import CreateProfileModel
+from src.users.schemas.user import CreateUserModel
 
 
 class AuthService:
-    def __init__(self, repository: AuthRepository):
+    def __init__(
+        self,
+        repository: UserRepository,
+    ):
         self.repository = repository
 
     def authenticate_user(
         self, username: str, password: str, db: Session
     ) -> Optional[User]:
+        """
+        Authenticates a user by checking the provided username and password against the database.
+
+        Args:
+        - username: The username of the user to authenticate.
+        - password: The password of the user to authenticate.
+        - db: The database session.
+
+        Returns:
+        - The authenticated user.
+        """
+        # Get user from the database by username
         user = self.repository.get_by_username(username=username, db=db)
+
+        # If user does not exist, raise UserDoesNotExist exception
         if not user:
             raise UserDoesNotExist
+
+        # If the provided password does not match the user's password, raise InvalidPassword exception
         if not verify_passwords(password, user.password):
             raise InvalidPassword
+
+        # Return the authenticated user
         return user
 
-    def register(self, user: UserCreate, db: Session) -> Optional[RegisterResponse]:
+    def create_user(self, user: CreateUserModel, db: Session):
+        """
+        Creates a new user in the database.
+
+        Args:
+            user (CreateUserModel): The user model to be created.
+            db (Session): The database session.
+
+        Returns:
+            CreateUserModel: The created user model.
+
+        Raises:
+            HTTPException: If the user already exists in the database.
+        """
+        # Hash the user's password
         user.password = hash_password(user.password)
+
+        try:
+            # Create the user in the database
+            user = self.repository.create_user(user=user, db=db)
+        except Exception:
+            # If the user already exists, raise an HTTPException
+            raise HTTPException(status_code=400, detail="User already exists")
+
+        return user
+
+    @staticmethod
+    def publish_user_created_event(
+        user_id: str, user: CreateUserModel, profile: CreateProfileModel
+    ) -> None:
+        """
+        Publishes a user created event to the cache handler.
+
+        Args:
+            user_id (str): The ID of the user.
+            user (CreateUserModel): The user object containing user details.
+            profile (CreateProfileModel): The profile object containing profile details.
+        """
+        # Publish user created event to the cache handler
```

Comment:
You should switch name to publish_user_event, becuase in the future maybe we will have more event than `created`

Reply:
but I want one method to one publisher it is more clear

Reply:
one event to one publisher

Reply:
Okey